client/app/lib/MyAccountsStore.js

```diff
@@ -71,26 +71,39 @@ export default class MyAccountsStore {
     let connector = this.find(c => c.id === connectorId)
     connector.accounts[accountIdx] = values
     return this.putConnector(connector)
-      .then(() => this.startConnectorPoll(connector.id))
   }
 
   synchronize (connectorId) {
     let connector = this.find(c => c.id === connectorId)
-    return this.putConnector(connector)
+    return this.importConnector(connector)
       .then(() => this.startConnectorPoll(connector.id))
   }
 
   deleteAccount (connectorId, accountIdx) {
     let connector = this.find(c => c.id === connectorId)
     connector.accounts.splice(accountIdx, 1)
-    return this.putConnector(connector)
+    return this.importConnector(connector)
       .then(() => this.updateConnector(connector))
   }
 
   putConnector (connector) {
     return this.fetch('PUT', `konnectors/${connector.id}`, connector)
       .then(response => {
         return response.status === 200
+          ? response.text()
+          : Promise.reject(response)
+      })
+      .then((body) => {
+        let connector = JSON.parse(body)
+        this.updateConnector(connector)
+        return connector
+      })
```

Comment:
I thought that was already the behaviour of _then_ : if a promise emit a _reject_, it's directly handled by the next _catch_ (or then 2nd argument).

So your proposition is exactly the same thing as the initial code.
Do I miss something ? : )